### DISCUSSION TITLE
how to create a multiaddr with custom protocol?

### DISCUSSION THREAD

**Discussion by Sherlock-Holo** [libp2p/rust-libp2p#3431] [Q&A]
currently the multiaddr is combined by some protocol, but the protocol is an enum so i can't create my protocol or my custom multiaddr with the custom protocol

is there a plan to allow user define a custom protocol?

**Comment by thomaseizinger** [comment]
We only re-export `Multiaddr` in this repo, this discussion is likely more fruitful over at https://github.com/multiformats/rust-multiaddr.

**Reply by thomaseizinger** [reply]
Afaik, the spec allows for "private" protocols with codes in a certain range. This could be better supported by the implementation I guess but so far, there wasn't any demand for it.
